homeassistant/components/systemmonitor/sensor.py

```diff
@@ -75,6 +67,17 @@ class SysMonitorSensorEntityDescription(SensorEntityDescription):
     """Description for System Monitor sensor entities."""
 
     mandatory_arg: bool = False
+    value_disk: Callable[[sdiskusage], float] | None = None
```

Comment:
Look at the renault sensor platform for how we can type the entity and entity description generic and have a single description method to calculate the native value.